src/main/webapp/app/course/learning-goals/learning-goal-card/learning-goal-card.component.html

```diff
@@ -30,6 +30,11 @@ <h4 class="m-0">
                 </div>
                 <ng-content></ng-content>
             </div>
+            <div *ngIf="!isPrerequisite && learningGoal.softDueDate" class="col-sm col py-2" style="max-width: fit-content">
+                <span style="vertical-align: middle"> {{ 'artemisApp.learningGoal.learningGoalCard.softDueDate' | artemisTranslate }} </span>
+                <span *ngIf="softDueDatePassed && !isMastered" class="badge bg-danger" id="date-badge-danger"> {{ learningGoal.softDueDate | artemisTimeAgo }} </span>
+                <span *ngIf="!(softDueDatePassed && !isMastered)" class="badge bg-success" id="date-badge-success"> {{ learningGoal.softDueDate | artemisTimeAgo }} </span>
```

Comment:
Since the 2 components are combined, I think the id can be removed here. Then the test assertion, which make use of the ids, need to be changed as well.